src/personnummer.js

```diff
@@ -1,246 +1,229 @@
+    this.age = '' + diffInYears(new Date(Date.now()), new Date(ageDate));
+
+    if (!this.#valid()) {
+      throw new PersonnummerError();
+    }
+  }
 
   /**
-   * Check if a Swedish social security number is for a female.
-   *
-   * @param {string|number} ssn
-   * @param {boolean} includeCoordinationNumber
-   *
-   * @throws Error when input value is not valid.
+   * Validate a Swedish social security number.
    *
    * @return {boolean}
    */
-  isFemale(ssn, includeCoordinationNumber = true) {
-    return !this.isMale(ssn, includeCoordinationNumber)
-  },
+  #valid() {
+    const valid = luhn(this.year + this.month + this.day + this.num) === +this.check && !!this.check;
+
+    if (valid && testDate(this.century + this.year, +this.month, +this.day)) {
+      return valid;
+    }
+
+    return valid && testDate(this.century + this.year, +this.month, (+this.day) - 60);
+  }
 
   /**
-   * Check if a Swedish social security number is for a male.
+   * Format a Swedish social security number as one of the official formats,
+   * A long format or a short format.
+   *
+   * If the input number could not be parsed a empty string will be returned.
    *
-   * @param {string|number} ssn
-   * @param {boolean} includeCoordinationNumber
+   * @param {boolean} longFormat
    *
    * @throws Error when input value is not valid.
    *
-   * @return {boolean}
+   * @return {string}
    */
-  isMale(ssn, includeCoordinationNumber = true) {
-    if (!this.valid(ssn, includeCoordinationNumber)) {
-      throw new Error('Invalid swedish social security number');
+  format(longFormat = false) {
+    if (longFormat) {
+      return `${this.century}${this.year}${this.month}${this.day}${this.num}${this.check}`;
     }
 
-    const parts = getParts(ssn);
-    const sexDigit = parts['num'].substr(-1);
+    return `${this.year}${this.month}${this.day}${this.sep}${this.num}${this.check}`;
+  }
 
-    return sexDigit % 2 === 1;
-  },
+  /**
+   * Check if a Swedish social security number is a coordination number or not.
+   *
+   * @return {boolean}
+   */
+  isCoordinationNumber() {
+    const day = +this.day
+    return day >= 61 && day < 91
+  }
 
   /**
-   * Validate a Swedish social security number.
+   * Check if a Swedish social security number is for a female.
    *
-   * @param {string|number} str
-   * @param {boolean} includeCoordinationNumber
+   * @throws Error when input value is not valid.
    *
    * @return {boolean}
    */
-  valid(ssn, includeCoordinationNumber = true) {
-    if (typeof ssn !== 'number' && typeof ssn !== 'string') {
-      return false;
-    }
-
-    const parts = getParts(ssn);
-    if (!Object.keys(parts).length) {
-      return false;
-    }
-
-    const valid = luhn(parts.year + parts.month + parts.day + parts.num) === +parts.check && !!parts.check;
-
-    if (valid && testDate(parts.century + parts.year, +parts.month, +parts.day)) {
-      return valid;
-    }
+  isFemale() {
```

Comment:
Could be a get instead of function, but this is okay too! :)

Reply:
Let's leave it since we should try to follow the spec :)